### PR TITLE
Chore: upgrade eslint

### PR DESCRIPTION
**Pull request by philipyoo**
I tried to separate my commits by each type of eslint change, so it might be easiest to view by each commit.

List of Eslint rules: https://eslint.org/docs/rules/

Note: Will squash commits after making any changes based on peoples' feedback

### CODE REVIEW

File: lib/IAMClient.js [hunk 1]
```diff
@@ -178,15 +178,15 @@ class VaultClient {
         };
         if (typeof options.marker !== 'undefined') {
             const marker = options.marker;
-            assert.notStrictEqual(isNaN(marker) || marker === '', true,
+            assert.notStrictEqual(Number.isNaN(marker) || marker === '', true,
```
Comment (philipyoo) [inline lib/IAMClient.js]: Test only breaks with this change 🤔 
Change of `Number.isNaN` below at L189 does not break tests

Comment (philipyoo) [reply lib/IAMClient.js]: TIL `Number.isNaN` is different from how `isNaN` works:
https://stackoverflow.com/questions/33164725/confusion-between-isnan-and-number-isnan-in-javascript

**Comment by philipyoo** [comment]
@ironman-machine try

**Comment by rahulreddy** [comment]
@alexandre-merle This is GTG.

**Pull request closed without merging**